app/services/opd_service/report_engine.rb

```diff
@@ -141,7 +141,7 @@ def registration(start_date, end_date)
           encounter_datetime as visit_date,
           p.birthdate,
           p.gender,
-          c.name AS visit_type
+          MIN(IFNULL(c.name, 'Unidentified')) AS visit_type
```

Comment:
`COALESCE` is recommended for better compatibility and code portability when it comes to MySQL. Not that there's any performance gains cause essentially they all do the same.